torch/csrc/distributed/autograd/engine/dist_engine.h

```diff
@@ -67,7 +67,7 @@ class TORCH_API DistEngine {
   // We also determine all leaf nodes(functions) in the graph and accumulate
   // them in outputEdges.
   void computeDependencies(
-      DistAutogradContext& context,
+      ContextPtr context,
```

Comment:
This can be a reference of the shared_ptr, as the call site always holds a copy of the `shared_ptr`?

Reply:
You can use a const ref. A const ref to a `shared_ptr` doesn't mean the wrapped object is const.